pychromecast/controllers/drtv.py

```diff
@@ -0,0 +1,118 @@
+"""Controller to interface with the DRTV app, from the Danish Broadcasting Corporation, dr.dk"""
+import threading
+import time
+import json
+
+from .media import STREAM_TYPE_BUFFERED, STREAM_TYPE_LIVE, MESSAGE_TYPE, TYPE_LOAD, BaseMediaPlayer
+from .. import __version__
+from ..config import APP_DRTV
+from ..error import PyChromecastError
+
+APP_NAMESPACE = "urn:x-cast:com.google.cast.media"
+
+class DRTVController(BaseMediaPlayer):
+    """Controller to interact with DRTV app."""
+
+    def __init__(self):
+        super().__init__(APP_DRTV)
+
+    def play_drtv(  # pylint: disable=too-many-locals
+        self,
+        media_id,
+        dr_session_tokens,
+        is_live=False,
+        current_time=0,
+        autoplay=True,
+        chainplay_countdown=10,
+        callback_function=None,
+    ):
+        """
+        Play DRTV media.
+
+        Parameters:
+        media_id: the id of the media to play, e.g. 20875
+        dr_session_tokens: JWT tokens to allow access to the content
+        chainplay_countdown: seconds to countdown before the next media in the chain (typically next episode) is played. -1 to disable
+        """
+        stream_type = STREAM_TYPE_LIVE if is_live else STREAM_TYPE_BUFFERED
+
+        session_tokens = json.loads(dr_session_tokens)
+        account_token = next((t for t in session_tokens if t['type'] == 'UserAccount'), {})
+        profile_token = next((t for t in session_tokens if t['type'] == 'UserProfile'), {})
+
+        msg = {
+            "media": {
+                "contentId": media_id,
+                "contentType": "video/hls",
+                "streamType": stream_type,
+                "metadata": {},
+                "customData": {
+                    "accessService": "StandardVideo"
+                },
+            },
+            MESSAGE_TYPE: TYPE_LOAD,
+            "currentTime": current_time,
+            "autoplay": autoplay,
+            "customData": {
+                "accountToken": account_token,
+                "chainPlayCountdown": chainplay_countdown,
+                "profileToken": profile_token,
+                "senderAppVersion": __version__,
+                "senderDeviceType": "pyChromeCast",
+                "showDebugOverlay": False,
+                "userId": ""
+            },
+        }
+
+        print(msg)
+        self.send_message(msg, inc_session_id=True, callback_function=callback_function)
+
+    def _get_drtokens(self):
```

Comment:
Guidance on whether this approach to retrieving tokens is acceptable is welcome :-)
It should work with Selenium as an optional dependency in this way.

Reply:
The stated lifetime of the tokens is 12 hours; there seems to be _some_ grace period, but I have been unable to ascertain how much.
I'm mostly worried about the user experience, if having to provide a hardcoded token that then regularly expires. Of course we could leave it up to the consumer of this API, i.e. somewhere inside Home Assistant, to do the token maintenance?

Reply:
Maybe code for getting the token etc. would fit in well in https://github.com/Eerovil/castextensions, @Eerovil wdyt?

Reply:
In the meantime I have pushed a commit to make the tokens a mandatory parameter in the controller, but left the Selenium way in the example code.
I also dug a bit deeper in the JWTs, and found a parameter in there indicating the tokens are valid for 1 month.